includes/wc-functions.php

```diff
@@ -1265,3 +1265,22 @@ function dokan_modify_vendor_order_counts( $counts ) {
 }
 
 add_filter( 'wp_count_posts', 'dokan_modify_vendor_order_counts', 10, 1 );
+
+
+/**
+ * @since DOKAN_SINCE
+ *
+ * @param boolean $is_purchasable
+ * @param object $product
+ *
+ * @return boolean
+ */
+function dokan_vendor_dokan_product_purchase_restriction( bool $is_purchasable, $product ): bool {
+    if ( dokan_is_product_author( $product->get_id() ) ) {
```

Comment:
method mane should be `dokan_vendor_self_product_purchase_restriction`

Reply:
since dokan_is_product_author() is doing db call an extra check will be more efficient:
eg:
if ( ! $is_purchasable ) {
return $is_purchasable;
}
then checking the regular if condition